share-coordinator/src/test/java/org/apache/kafka/coordinator/share/ShareCoordinatorShardTest.java

```diff
@@ -791,6 +793,304 @@ public void testNonSequentialBatchUpdates() {
         verify(shard.getMetricsShard(), times(3)).record(ShareCoordinatorMetrics.SHARE_COORDINATOR_WRITE_SENSOR_NAME);
     }
 
+    @Test
+    public void testStateBatchCombine() {
```

Comment:
Might be better if you parametrize the test with thest input you created.

Reply:
Parameterization will again result in MethodLength issue as the tests need to be placed in a generator method.

Reply:
But parametrization will not fail specific scenario which is often easy to debug and fix.

Reply:
Ok, will add